soc/arm/st_stm32/common/Kconfig.soc

```diff
@@ -21,3 +21,12 @@ config USE_STM32_ASSERT
 	bool "STM32Cube HAL and LL drivers asserts"
 	help
 	  Enable asserts in STM32Cube HAL and LL drivers.
+
+if SOC_SERIES_STM32H7X
+
+config POWER_SUPPLY_SMPS
```

Comment:
@erwango, other STM32 series have LDO/SMPS (even if not yet supported in Zephyr), for example stm32wl, stm32u5, ...
So it may be good to keep this swich in a common Kconfig.

Reply:
@erwango advice if what ABOSTM comments is rational else I will make the amendments

Reply:
ok, let's keep it here then.
But, please use more compact "depends on", when only one symbol is impacted (instead of "if FOO").

Reply:
Done

Reply:
Would you mind pushing this in the initial commit ?

Reply:
Done